test/tools/router_check/test/router_test.cc

```diff
@@ -0,0 +1,121 @@
+#include <string>
+
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+#include "test/tools/router_check/router.h"
+#include "test/tools/router_check/validation.pb.h"
+
+#include "absl/strings/str_join.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+constexpr char kDir[] = "test/tools/router_check/test/config/";
```

Comment:
Please use [temporaryPath](https://github.com/envoyproxy/envoy/blob/266c9fa51b0d1dbce746abda6c1b47b3ca19c3cb/test/test_common/environment.h#L101), and `removePath` upon tear-down

Reply:
I missed that it also holds files, I thought it only creates files there.
I would suggest not creating temp files in non-temp directories as part of tests. This avoids situations where a stale file (e.g., created and not deleted because the test stopped mid-way) interferes with future executions of the tests.